setup.py

```diff
@@ -766,7 +765,7 @@ def setup_package():
         long_description=LONG_DESCRIPTION,
         classifiers=CLASSIFIERS,
         platforms="any",
-        python_requires=">=3.6.1",
+        python_requires=">=3.7.1",
```

Comment:
```suggestion
        python_requires=">=3.7.0",
```

I think the patch bump before was for some critical mypy fixes; we can probably start clean at 3.7